src/libgit2/config.c

```diff
@@ -606,8 +607,19 @@ static int get_backend_for_use(git_config_backend **out,
 
 	git_vector_foreach(&cfg->backends, i, backend) {
 		if (!backend->backend->readonly) {
-			*out = backend->backend;
-			return 0;
+			switch (use) {
+			case BACKEND_USE_SET:
+				*out = backend->backend;
+				return 0;
+			case BACKEND_USE_DELETE:
+				/* return the first backend that contains the given config-name */
+				backend->backend->get(backend->backend, name, &entry);
+				if (entry != NULL) {
+					*out = backend->backend;
+					return 0;
+				}
```

Comment:
If fixing the pipeline is that simple, I'll give you a beer emoji. I'll try that right now.

Reply:
I'm still going to try it (what's the worst that can happen?) but I don't think this will resolve the issue given that we weren't freeing `entry` before:

```diff
@@ -606,8 +607,19 @@ static int get_backend_for_use(git_config_backend **out,
 
 	git_vector_foreach(&cfg->backends, i, backend) {
 		if (!backend->backend->readonly) {
-			*out = backend->backend;
-			return 0;
+			switch (use) {
+			case BACKEND_USE_SET:
+				*out = backend->backend;
+				return 0;
+			case BACKEND_USE_DELETE:
+				/* return the first backend that contains the given config-name */
+				backend->backend->get(backend->backend, name, &entry);
+				if (entry != NULL) {
+					*out = backend->backend;
+					return 0;
+				}
+				break;
+			}
 		}
 	}
```